src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs

```diff
@@ -513,6 +535,57 @@ public HashSet<Guid> Dependencies
             }
         }
 
+        /// <summary>
+        /// Gathers the packages that this graph depends on
+        /// </summary>
+        public IEnumerable<PackageInfo> PackageDependencies
+        {
+            get
+            {
+                // Get set of assemblies used by nodes in this workspace
+                var assembliesUsed = GetAssembliesUsed();
+                
+                // Collect pacakge dependencies of nodes in workspace
+                var packageDependencies = new HashSet<PackageInfo>();
+                if (CollectingNodePackageDependencies != null)
+                {
+                    var npd = CollectingNodePackageDependencies(assembliesUsed);
+                    foreach (var p in npd)
+                    {
+                        packageDependencies.Add(p);
+                    }
+                }
+
+                // Add custom node package dependencies
+                var customNodes = Nodes.Where(node => node.GetType() == typeof(Function));
+                var customNodeIDs = customNodes.Select(node => (node as Function).Definition.FunctionId);
+                if (CollectingCustomNodePackageDependencies != null)
+                {
+                    var cnpd = CollectingCustomNodePackageDependencies(customNodeIDs);
+                    foreach (var p in cnpd)
+                    {
+                        packageDependencies.Add(p);
+                    }
+                }
+
+                return packageDependencies;
+            }
+        }
+
+        /// <summary>
+        /// Gathers the assemblies that are used by nodes in this workspace
+        /// </summary>
+        /// <returns></returns>
+        internal HashSet<AssemblyName> GetAssembliesUsed()
```

Comment:
can you make this name and comment more explicit, it doesn't gather the full tree of dependencies `used` by the graph, it gathers the top level references of those packages that are present in the graph.

Reply:
Changed to `GetAssembliesReferencedByNodes`